assets/scripts/app/ScrollIndicators.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { registerKeypress } from './keypress'
+
+class ScrollIndicators extends React.Component {
+  static propTypes = {
+    posLeft: PropTypes.number.isRequired,
+    posRight: PropTypes.number.isRequired,
+    scrollStreet: PropTypes.func.isRequired,
+    scrollTop: PropTypes.number.isRequired
+  }
+
+  shouldComponentUpdate (nextProps) {
+    if (this.props.posLeft !== nextProps.posLeft &&
+        this.props.posRight !== nextProps.posRight) {
+      this.refs.left_indicator.innerHTML = Array(nextProps.posLeft + 1).join('‹')
+      this.refs.right_indicator.innerHTML = Array(nextProps.posRight + 1).join('›')
+      return true
+    }
+    // Checking if window was resized
+    if (this.props.scrollTop !== nextProps.scrollTop) {
+      this.updateIndicatorsPosition(nextProps.scrollTop)
+      return true
+    }
+    return false
+  }
+
+  componentDidMount () {
+    registerKeypress('left', (event) => { this.props.scrollStreet(true, event.shiftKey) })
+    registerKeypress('right', (event) => { this.props.scrollStreet(false, event.shiftKey) })
```

Comment:
For the callback function here, and for the respective `onClick` handler on the elements themselves, we should set up a method on this component so that the handlers share code.

Reply:
@louh sorry, what do you mean by 'share code'? 
Should I add a method, say `handleScroll`  in the `ScrollIndicators` component that calls `this.props.scrollStreet()` and then pass that to `registerKeypress`?
So... `registerKeypress('left', this.handleScroll)`?

Reply:
Yes, just like that! You will need two of them, one for handling going left, and one for handling going right.